src/random.cc

```diff
@@ -3,7 +3,6 @@
 
 #include "containers/scoped.hpp"
 #include "thread_local.hpp"
-#include <memory>
```

Comment:
We don't want to remove this include -- we use std::unique_ptr in this file.  zouyonghao might have added it because it was necessary to compile, on some platform whose stdlib does not include <memory> via one of the other included headers.  So we should keep it.

Also, generally, it is preferable to have include lines in the same file for any library feature used in that file, even if a header already has it. One reason for this is that it makes it easier to remove include lines from frequently used header files (because there will later be fewer .cc files that need to be updated). That would reduce compilation time.  (This is a very unimportant, weak benefit.)

Reply:
It turned out that compilation in Jammy fails without this include. I’m going to put this back.